homeassistant/components/nice_go/switch.py

```diff
@@ -0,0 +1,49 @@
+"""Nice G.O. switch platform."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from homeassistant.components.switch import SwitchDeviceClass, SwitchEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import NiceGOConfigEntry
+from .entity import NiceGOEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: NiceGOConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Nice G.O. switch."""
+    coordinator = config_entry.runtime_data
+
+    async_add_entities(
+        NiceGOSwitchEntity(coordinator, device_id, device_data.name, "switch")
```

Comment:
There's no need to add the platform domain to the unique id of the entity. The entity registry is aware of both the integration domain and the platform domain of the entity.

What we do need to do is to identify entities uniquely that come from the same platform and the same device. This usually comes in to play especially for the sensor platform.